common/session/class.BasicSession.php

```diff
@@ -142,6 +142,10 @@ public function getDataLanguage()
      */
     public function getInterfaceLanguage()
     {
+        if (PHPSession::singleton()->hasAttribute('overrideInterfaceLanguage')) {
+            return PHPSession::singleton()->getAttribute('overrideInterfaceLanguage');
+        }
+
         return $this->getServiceLocator()->get(UserLanguageServiceInterface::class)->getInterfaceLanguage($this->getUser());
```

Comment:
WDYT of adding that overriding value check on the `UserLanguageServiceInterface` [implementation](https://github.com/oat-sa/generis/blob/master/common/oatbox/user/UserLanguageService.php#L57) level?

You can then easily extend [the test cases](https://github.com/oat-sa/generis/blob/master/test/unit/oatbox/user/UserLanguageServiceTest.php#L43) covering that method. You will also ensure the interface language is being overridden consistently, as some other pieces of code might be using the `UserLanguageService` directly.

Reply:
Good idea, I have isolated the logic into the UserLanguageService implementation and use the session to control when it should be set to avoid having a hard dependency with the session.